container/lxd/server.go

```diff
@@ -63,6 +63,16 @@ func NewLocalServer() (*Server, error) {
 	return svr, errors.Trace(err)
 }
 
+// NewRemoteServer returns a Server based on a remote connection.
+func NewRemoteServer(remote RemoteServer) (*Server, error) {
```

Comment:
The nomenclature might get a little confusing here. We have ```NewRemoteServer``` that is not returning a ```RemoteServer``` instance as might be expected.

Maybe a better name for ```RemoteServer``` is something like ```ServerSpec```. And we could abstract away dealing with the upstream ```ConnectionArgs``` type by have a ```NewServerSpec``` method that takes name, host, protocol, client cert, server cert.

Reply:
Agreed, in fact this is what was confusing with the old `tools/lxdclient` and hard to prise apart.